ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java

```diff
@@ -280,5 +280,9 @@ private void throwIfInsertOnReadOnlyTopic(
       throw new KsqlException("Cannot insert into read-only topic: "
           + dataSource.getKafkaTopicName());
     }
+
+    if (!dataSource.getSchema().headers().isEmpty()) {
+      throw new KsqlException("Cannot insert into a source with header columns");
```

Comment:
Instead of `source`, use the name of the source. The `dataSource.getDataSourceType()` returns the type.